system/jlib/jfile.hpp

```diff
@@ -159,6 +159,33 @@ extern jlib_decl unsigned sortDirectory(
 
 
 
+enum FileIOStat {
+    FIOSiocycles,
+    FIOSiobytes,
+    FIOSmax
+};
+
+class jlib_decl FileIOStats
+{
+public:
+    FileIOStats() { _clear(values); }
+
+    void set(FileIOStat stat, unsigned __int64 value) { values[stat] = value; }
+    void merge(FileIOStat stat, unsigned __int64 value) { values[stat] += value; }
+    void merge(IFileIO * io);
+    void merge(const FileIOStats & other);
+    void merge(MemoryBuffer & in);
+    inline unsigned __int64 getStatistic(FileIOStat stat) const { return values[stat]; }
+    void deserialize(MemoryBuffer & in);
+    //void serialize(MemoryBuffer & out) const;
+    void serializeNormalized(MemoryBuffer & out) const; // Also converts cycles to nanoseconds.
```

Comment:
This is the one part of the implementation I'm not so happy about.  Any bright ideas welcome.